source/extensions/clusters/redis/redis_cluster.cc

```diff
@@ -351,29 +453,78 @@ void RedisCluster::RedisDiscoverySession::onResponse(
       return;
     }
 
-    // Field 2: Primary address for slot range
-    auto primary_address = ProcessCluster(slot_range[SlotPrimary]);
-    if (!primary_address) {
+    // Row 3: Primary slot address
+    if (!validateCluster(slot_range[SlotPrimary])) {
       onUnexpectedResponse(value);
       return;
     }
+    // Try to parse primary slot address as IP address
+    // It may fail in AWS ElastiCache use case: it uses hostnames instead of IPs.
```

Comment:
I'd suggest avoiding talking about the specific AWS use case, and instead talk about generally supporting hostnames and perhaps listing ElastiCache as an example of that